src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -255,6 +262,11 @@ public String getName() {
     return Collections.unmodifiableMap(resourceUrls);
   }
 
+  @NotNull
+  public CoursePluginVersion getRequiredPluginVersion() {
+    return minimumPluginVersion;
+  }
```

Comment:
The name of the variable and the getter are pretty confusing. With this name, I would assume the version to correspond to a plugin version (e.g. 2.1), but it's actually a separate version for the course itself. Perhaps something like `courseVersion`?

Reply:
`courseVersion` is nice, but I still opt for including the word "plugin" in the JSON property name, since in the end it's truly a "course version" only in the context of our plugin

Reply:
I could see someone creating the course config seeing `"minPluginVersion"` and going "oh right, I want my students to use at least version 2.2 of the plugin" :thinking:

Reply:
Will change to `version`